lib/OnyxUtils.ts

```diff
@@ -1251,12 +1249,131 @@ function doAllCollectionItemsBelongToSameParent<TKey extends CollectionKeyBase>(
+            // since there are none matched. In withOnyx() we wait for all connected keys to return a value before rendering the child
+            // component. This null value will be filtered out so that the connected component can utilize defaultProps.
+            if (matchingKeys.length === 0) {
+                if (mapping.key && !isCollectionKey(mapping.key)) {
+                    cache.addNullishStorageKey(mapping.key);
+                }
+
+                // Here we cannot use batching because the nullish value is expected to be set immediately for default props
+                // or they will be undefined.
+                sendDataToConnection(mapping, null, undefined, false);
+                return;
+            }
+
+            // When using a callback subscriber we will either trigger the provided callback for each key we find or combine all values
+            // into an object and just make a single call. The latter behavior is enabled by providing a waitForCollectionCallback key
+            // combined with a subscription to a collection key.
+            if (typeof mapping.callback === 'function') {
+                if (isCollectionKey(mapping.key)) {
+                    if (mapping.waitForCollectionCallback) {
+                        getCollectionDataAndSendAsObject(matchingKeys, mapping);
+                        return;
+                    }
+
+                    // We did not opt into using waitForCollectionCallback mode so the callback is called for every matching key.
+                    multiGet(matchingKeys).then((values) => {
+                        values.forEach((val, key) => {
+                            sendDataToConnection(mapping, val as OnyxValue<TKey>, key as TKey, true);
+                        });
+                    });
+                    return;
+                }
+
+                // If we are not subscribed to a collection key then there's only a single key to send an update for.
+                get(mapping.key).then((val) => sendDataToConnection(mapping, val as OnyxValue<TKey>, mapping.key, true));
+                return;
+            }
+
+            // If we have a withOnyxInstance that means a React component has subscribed via the withOnyx() HOC and we need to
+            // group collection key member data into an object.
+            if ('withOnyxInstance' in mapping && mapping.withOnyxInstance) {
+                if (isCollectionKey(mapping.key)) {
+                    getCollectionDataAndSendAsObject(matchingKeys, mapping);
+                    return;
+                }
+
+                // If the subscriber is not using a collection key then we just send a single value back to the subscriber
+                get(mapping.key).then((val) => sendDataToConnection(mapping, val as OnyxValue<TKey>, mapping.key, true));
+                return;
+            }
+
+            console.error('Warning: Onyx.connect() was found without a callback or withOnyxInstance');
```

Comment:
Maybe this should throw instead?

Reply:
Hmm this log was already in this function before my PR -- maybe we could remove it but I would like to do this in another PR to reduce the chance of regressions. WDYT?

Reply:
Sure, that sounds fine with me.